### ISSUE TITLE
OSC API design makes it difficult to use SoundScaper with OSC Timelines applications

### ISSUE THREAD

**Issue by onar3d** [felixfaire/Soundscaper-OSC#4]
Hi!

I tested using SoundScaper with TWO, and found that its OSC API makes it difficult to use that way.

[The longer discussion can be found in the TWO forum.](https://forum.controlmedia.art/t/two-crashes-on-timeline-stop-recording/88/6).

In a few words, the problem is that all OSC messages require the "voice ID" as an value in the OSC message, rather than it being part of the OSC address.

If this was changed, so that:

'/update' <voice-id> <x> <y> <z>

would instead become:

'voice1/update'  <x> <y> <z>

'voice2/update'  <x> <y> <z>

etc.

Across all messages.

This way, the voice namespace can be standard (/update, / start, etc, with the root differing depending on the root), and the actual message payload, e.g. XYZ position, can be separate from the address/target, meaning e.g. an OSC application with an XYZ position widget, can be used to control voices.

You don't need to replace your existing scheme with the one I suggest either - you can support both in parallel too, thus maintaining backwards compatibility.

Feel free to contact me if you want to collaborate more closely on getting TWO and Soundscaper interacting well!

I've done something similar with a wide range of OSC-capable media apps over the years, commercial and open-source alike.

With that said, Soundscaper will work better also with other OSC sequencer apps, e.g. Vezer, with the change I suggest.

Even more if it also then supports a discovery protocol, e.g. OSC Query, but that's a whole new feature and not needed for the above!
